### PR TITLE
[KOGITO-9218] - Update SonataFlowPlatform struct

### PR DESCRIPTION
**Pull request by ricardozanini**
**JIRA:**
https://issues.redhat.com/browse/KOGITO-9218

**Description:**
Follow up on the changes on:
- https://github.com/kiegroup/kogito-serverless-operator/pull/214

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributions doc](https://github.com/kiegroup/kogito-docs/blob/main/CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [ ] The nav.adoc file has a link to this guide in the proper category
- [ ] The index.adoc file has a card to this guide in the proper category, with a meaningful description

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Review by MarianMacik — APPROVED** [review]
Looks good to me.

**Comment by ricardozanini** [comment]
@MarianMacik mind taking another look? I've added a migration guide for the operator since we are changing the YAML structure.

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/build-and-deploy-workflows.adoc
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/build-and-deploy-workflows.adoc:231]:
```suggestion
The `spec.build.config.strategyOptions.KanikoBuildCacheEnabled` enables the internal Kaniko process to cache builder images to try to speed up the building execution.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/migration-guide.adoc
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/migration-guide.adoc:5]:
```suggestion
When migrating to version 1.43.0, you must review the `SonataFlowPlatform` instances.
```

**Review by kaldesai — APPROVED** [review]
@ricardozanini Minor suggestion. Otherwise, your content looks good to me. Thank you!

**Review by MarianMacik — APPROVED** [review]
@ricardozanini Changes look good after migration guide inclusion.

**Review by radtriste — APPROVED** [review]

**Comment by radtriste** [comment]
@ricardozanini I added a card to the `cloud/index.adoc`
Feel free to merge if it is ok

**Comment by ricardozanini** [comment]
Thank you folks!

I'm putting this on hold until we merge the operator's PR.

**Pull request merged**
